src/libraries/System.Net.Http/src/System/Net/Http/HttpTelemetry.cs

```diff
@@ -50,18 +50,24 @@ public void RequestStart(HttpRequestMessage request)
                 request.VersionPolicy);
         }
 
-        [Event(2, Level = EventLevel.Informational)]
-        public void RequestStop()
+        [NonEvent]
+        public void RequestStop(HttpResponseMessage? response)
+        {
+            RequestStop(response is null ? -1 : (int)response.StatusCode);
+        }
+
+        [Event(2, Level = EventLevel.Informational, Version = 1)]
+        private void RequestStop(int statusCode)
         {
             Interlocked.Increment(ref _stoppedRequests);
-            WriteEvent(eventId: 2);
+            WriteEvent(eventId: 2, statusCode);
         }
 
-        [Event(3, Level = EventLevel.Error)]
-        public void RequestFailed()
+        [Event(3, Level = EventLevel.Error, Version = 1)]
+        public void RequestFailed(string exception)
```

Comment:
Would it make sense to have `RequestFailed` as `NonEvent` (similarly to `RequestStop`) and pass in the `Exception`? Then the decision of what exception details are logged would be within the telemetry class and not on the caller side.

Reply:
Sure, I'll tweak the layering